bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -108,183 +103,183 @@ public void handleCommand(ChannelUID channelUID, Command command) {
-            reconnectTask = scheduler.schedule(this::connect, conf.connectionRetryInterval, TimeUnit.SECONDS);
+            if (handler != null) {
+                handler.receiveMessage(message);
+            }
         }
     }
 
     @Override
-    public void receiveMessage(Message message) {
-        if (DataItemGroup.dataItemGroups.containsKey(message.getID())) {
-            DataItem[] dataItems = DataItemGroup.dataItemGroups.get(message.getID());
-
-            for (DataItem dataItem : dataItems) {
-                String channelId = dataItem.getSubject();
-
-                if (!OpenThermGatewayBindingConstants.SUPPORTED_CHANNEL_IDS.contains(channelId)
-                        || (dataItem.getFilteredCode() != null && dataItem.getFilteredCode() != message.getCode())) {
-                    continue;
-                }
-
-                State state = null;
-
-                switch (dataItem.getDataType()) {
-                    case FLAGS:
-                        state = OnOffType.from(message.getBit(dataItem.getByteType(), dataItem.getBitPos()));
-                        break;
-                    case UINT8:
-                    case UINT16:
-                        state = new DecimalType(message.getUInt(dataItem.getByteType()));
-                        break;
-                    case INT8:
-                    case INT16:
-                        state = new DecimalType(message.getInt(dataItem.getByteType()));
-                        break;
-                    case FLOAT:
-                        float value = message.getFloat();
-                        @Nullable
-                        Unit<?> unit = dataItem.getUnit();
-                        state = (unit == null) ? new DecimalType(value) : new QuantityType<>(value, unit);
-                        break;
-                    case DOWTOD:
-                        break;
-                }
+    public void connectionStateChanged(ConnectionState state) {
+        scheduler.submit(() -> connectionStateChangedTask(state));
+    }
 
-                if (state != null) {
-                    logger.debug("Received update for channel '{}': {}", channelId, state);
-                    updateState(channelId, state);
-                }
+    private void connectionStateChangedTask(ConnectionState state) {
+        if (this.state != state) {
+            this.state = state;
+
+            switch (state) {
+                case CONNECTED:
+                    updateStatus(ThingStatus.ONLINE);
+                    cancelAutoReconnect();
+                    break;
+                case DISCONNECTED:
+                    if (!disposing) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                        autoReconnect();
+                    }
+                    break;
+                default:
+                    updateStatus(ThingStatus.UNKNOWN);
```

Comment:
Is this needed? If I understand it correctly (which I may not), this method will be called with `ConnectionState.CONNECTING` when trying to connect, which could be for transitioning from **OFFLINE** back to **ONLINE**. In this case it probably wouldn't make much sense to switch to **UNKNOWN** briefly during this transition? On the other hand, before leaving `initialize()`, **UNKNOWN** should be set.

Reply:
> wouldn't make much sense to switch to UNKNOWN briefly 

Ok. Done.

> before leaving initialize(), UNKNOWN should be set.

Ok. Done.